### PR TITLE
[front] - fix: update URL validation to allow @ character

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR ensures that the "@" character is now allowed in url when creating a website.

### CODE REVIEW

**Review by Fraggle — APPROVED** [review]

File: front/lib/webcrawler.ts [hunk 1]
```diff
@@ -32,11 +32,11 @@ export function urlToDataSourceName(url: string) {
 
 const urlValidationRegex = new RegExp(
   "^(https?:\\/\\/)?" + // validate protocol
-    "((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|" + // validate domain name
+    "((([a-z\\d@]([a-z\\d@-]*[a-z\\d@])*)\\.)+[a-z]{2,}|" + // validate domain name with @ allowed
     "((\\d{1,3}\\.){3}\\d{1,3}))" + // validate OR ip (v4) address
-    "(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*" + // validate port and path
-    "(\\?[;&a-z\\d%_.~+=-]*)?" + // validate query string
-    "(\\#[-a-z\\d_]*)?$",
+    "(\\:\\d+)?(\\/[-a-z\\d%@_.~+]*)*" + // validate port and path
```
Comment (Fraggle) [inline front/lib/webcrawler.ts:37]: i'm not 100% sure here.

Comment (JulesBelveze) [reply front/lib/webcrawler.ts:37]: @Fraggle should we just use `https://github.com/dust-tt/dust/blob/52af1a6e87172d0367e4b9ab6dcd50f721afd47d/types/src/shared/utils/url_utils.ts`?

Comment (Fraggle) [reply front/lib/webcrawler.ts:37]: why not :)

Comment (JulesBelveze) [reply front/lib/webcrawler.ts]: @Fraggle this should at least be more homogeneous 😄

Comment (Fraggle) [reply front/lib/webcrawler.ts]: and we love homogeneous things :)

**Pull request merged**